src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerConfigBuilder.java

```diff
@@ -0,0 +1,178 @@
+
+    private static final double DEFAULT_FAILURE_RATE_THRESHOLD = 0.8;
+
+    private static final Scope DEFAULT_SCOPE = Scope.SERVICE;
+
+    private static final ExceptionFilter DEFAULT_FAILURE_FILTER = cause -> true;
+
+    private static final long DEFAULT_MINIMUM_REQUEST_THRESHOLD = 10;
+
+    private static final Duration DEFAULT_TRIAL_REQUEST_INTERVAL = Duration.ofSeconds(3);
+
+    private static final Duration DEFAULT_CIRCUIT_OPEN_WINDOW = Duration.ofSeconds(10);
+
+    private static final Duration DEFAULT_COUNTER_SLIDING_WINDOW = Duration.ofSeconds(20);
+
+    private static final Duration DEFAULT_COUNTER_UPDATE_INTERVAL = Duration.ofSeconds(1);
+
+    private final String remoteServiceName;
+
+    private double failureRateThreshold = DEFAULT_FAILURE_RATE_THRESHOLD;
+
+    private Scope scope = DEFAULT_SCOPE;
+
+    private ExceptionFilter exceptionFilter = DEFAULT_FAILURE_FILTER;
+
+    private long minimumRequestThreshold = DEFAULT_MINIMUM_REQUEST_THRESHOLD;
+
+    private Duration trialRequestInterval = DEFAULT_TRIAL_REQUEST_INTERVAL;
+
+    private Duration circuitOpenWindow = DEFAULT_CIRCUIT_OPEN_WINDOW;
+
+    private Duration counterSlidingWindow = DEFAULT_COUNTER_SLIDING_WINDOW;
+
+    private Duration counterUpdateInterval = DEFAULT_COUNTER_UPDATE_INTERVAL;
+
+    /**
+     * Creates a new {@link CircuitBreakerConfigBuilder} with the specified name of remote service.
+     *
+     * @param remoteServiceName The remote service name to be logged
+     */
+    public CircuitBreakerConfigBuilder(String remoteServiceName) {
+        this.remoteServiceName = requireNonNull(remoteServiceName);
+        if (remoteServiceName.isEmpty()) {
+            throw new IllegalArgumentException("remoteServiceName must not be empty");
+        }
+    }
+
+    /**
+     * Sets the threshold of failure rate to detect a remote service fault.
+     *
+     * @param failureRateThreshold The rate between 0 (exclusive) and 1 (inclusive)
+     */
+    public CircuitBreakerConfigBuilder failureRateThreshold(double failureRateThreshold) {
+        if (failureRateThreshold <= 0 || 1 < failureRateThreshold) {
+            throw new IllegalArgumentException(
+                    "failureRateThreshold must be between 0 (exclusive) and 1 (inclusive)");
+        }
+        this.failureRateThreshold = failureRateThreshold;
+        return this;
+    }
+
+    /**
+     * Sets the {@link Scope} indicating the policy of circuit breaker scoping.
+     */
+    public CircuitBreakerConfigBuilder scope(Scope scope) {
+        this.scope = requireNonNull(scope, "scope");
+        return this;
+    }
+
+    /**
+     * Sets the minimum number of requests within a time window necessary to detect a remote service fault.
+     */
+    public CircuitBreakerConfigBuilder minimumRequestThreshold(long minimumRequestThreshold) {
+        if (minimumRequestThreshold < 0) {
+            throw new IllegalArgumentException("minimumRequestThreshold must be >= 0");
```

Comment:
Global comment: please make the exception messages consistent with other exception messages in Armeria, and also add the specified value to the exception message:

    "minimumRequestThreshold: " + minimumRequestThreshold + " (expected: >= 0)"